### PR TITLE
feat(auto-instrumentation-node): add undici instrumentation

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- auto-instrumentations-node is expected to include all instrumentations in js and js-contrib, but didn't yet have the newly created and published undici

## Short description of the changes

- add undici instrumentation to auto-instrumentations-node package

Note: As I understand it, we haven't had an official "release" of undici instrumentation, but we recently found that new packages like this still [got published on npm](https://www.npmjs.com/package/@opentelemetry/instrumentation-undici) due to a [misconfiguration in our workflows](https://github.com/open-telemetry/opentelemetry-js-contrib/issues/2020). 

Also, I couldn't recall if there was a reason to delay adding this to the package (like since it's still so new), but figured I'd open the PR either way because I didn't see it otherwise.

### CODE REVIEW

File: metapackages/auto-instrumentations-node/README.md
Comment (trentm) [inline metapackages/auto-instrumentations-node/README.md:181]: My alphabetical spidey-sense tingled here: I think this list is missing instr-tedious.
Anyway, obviously not something this PR needs worry about.

**Review by trentm — APPROVED** [review]

**Review by david-luna — APPROVED** [review]

**Review by blumamir — APPROVED** [review]

**Comment by blumamir** [comment]
Thanks @JamieDanielson , can you please resolve the conflicts so this can merge?

File: metapackages/auto-instrumentations-node/README.md
Comment (JamieDanielson) [reply metapackages/auto-instrumentations-node/README.md:183]: Oof good catch! Yes that's a good follow-up though, will take a look once this merges. At least the instrumentation is in there so that's probably the most important part 😄

**Comment by JamieDanielson** [comment]
I think this is good to go, but unsure if I should hold off for a maintainer to merge as they are assigned component owners (I am but a lowly approver 😃 )

**Comment by blumamir** [comment]
> I think this is good to go, but unsure if I should hold off for a maintainer to merge as they are assigned component owners (I am but a lowly approver 😃 )

Not sure about the formalities, but since it already has 3 approves and is open for more than a week, I think it's safe to merge 👍

**Pull request merged**
